### PR TITLE
DISP-3444: Add auth token support + auto refresh bearer token

### PR DESCRIPTION
**Pull request by jraede**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+machine:
+  python:
+    version: 2.7.6
+cache:
+  directories:
+    - $HOME/.npm
+
+install:
+  - npm install
```
Comment (aldeed) [inline .travis.yml]: You don't need the install/script. Travis runs `npm install` and `npm test` automatically.

https://docs.travis-ci.com/user/languages/javascript-with-nodejs#Default-Test-Script
https://docs.travis-ci.com/user/languages/javascript-with-nodejs#Travis-CI-uses-npm

File: lib/dispatch.js [hunk 2]
```diff
@@ -79,6 +82,16 @@ export default class Dispatch {
   }
 
   /**
+   * Log in with a limited-use auth token retrieved via /v1/auth_tokens
+   * @param  {String} authToken
+   * @return {Promise}
+   */
+  loginAuthToken(authToken) {
+    return this.getNoAuthClient().post(`/v1/auth_tokens/${authToken}/exchange`)
+    .then(response => this.handleBearerToken(response.access_token, response.refresh_token));
```
Comment (aldeed) [inline lib/dispatch.js]: `.then` line should be indented

File: lib/dispatch.tests.js [hunk 3]
```diff
@@ -65,6 +69,21 @@ describe('client SDK', () => {
...
+    it('should log in with auth token', done => {
+      const req = nock('https://api.dispatch.me').post('/v1/auth_tokens/12345/exchange').reply(200, {
+        access_token: testBearerToken,
+        refresh_token: testRefreshToken,
+      });
+
+      const client = new Dispatch(testClientID, testClientSecret);
+      client.loginAuthToken('12345').then(token => {
+        expect(token).toEqual(testBearerToken);
+        expect(client.bearerToken).toEqual(testBearerToken);
```
Comment (aldeed) [inline lib/dispatch.tests.js]: Do you need to test that the refresh token is also updated?

Comment (aldeed) [inline lib/dispatch.tests.js]: nm, I see that tested elsewhere

File: lib/errors.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+function makeError(klass) {
+  klass.prototype = Object.create(Error.prototype);
+}
+export class UnauthorizedError {
```
Comment (aldeed) [inline lib/errors.js]: Doesn't `UnauthorizedError extends Error` work rather than `makeError`?

File: package.json [hunk 5]
```diff
@@ -4,7 +4,7 @@
   "description": "High- and low-level libraries for interacting with the Dispatch API",
   "main": "dist/lib/index.js",
   "scripts": {
-    "test": "mocha --compilers js:babel-core/register --recursive 'lib/**/*.tests.js'",
+    "test": "npm run lint && node_modules/.bin/mocha --compilers js:babel-core/register --recursive 'lib/**/*.tests.js'",
```
Comment (aldeed) [inline package.json]: I think it should just be `mocha`. It automatically looks in `node_modules` before global.

File: lib/errors.js (hunk 4 above)
Comment (jraede) [inline lib/errors.js]: No there were some issues with extending native classes in Babel and from what I read it's generally frowned upon. This is apparently the fool-proof way to get everything to work.

File: package.json (hunk 5 above)
Comment (jraede) [inline package.json]: Oh, didn't realize that. Will change.

**Comment by aldeed** [comment]
LGTM

**Pull request merged**